gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MfaFilterContext.java

```diff
@@ -131,6 +136,34 @@ public boolean isDeviceRiskAssessmentEnabled() {
     public boolean isChallengeOnceCompleted() {
         return TRUE.equals(session.get(MFA_CHALLENGE_COMPLETED_KEY));
     }
+
+
+    public void setDefaultFactorWhenApplied() {
+        String enrollingFactorId = session.get(ENROLLED_FACTOR_ID_KEY);
+        FactorSettings factorSettings = client.getFactorSettings();
+        if (nonNull(enrollingFactorId) && nonNull(factorSettings) && !matchFactorRule(enrollingFactorId, factorSettings)) {
```

Comment:
question: why are we overriding the ENROLLED_FACTOR_ID_KEY into the session with the default if the session already have a ENROLLED_FACTOR_ID_KEY

Reply:
we have a story for this. Let's discuss and continue within [AM-1138](https://gravitee.atlassian.net/browse/AM-1138)